autonat/autonat-v2.md

```diff
@@ -95,25 +104,23 @@ the message. This includes inability to decode the requested address to dial.
 peer that prevented it from completing the request.
 
 
-### Consideration for DDOS Prevention
+### Amplification Attack Prevention
+
+When a client asks a server to dial an address that is not the clients observed
+ip address, the server asks the client to send him some non trivial amount of
+bytes as a cost to dial a different ip address. The number of bytes is decided
+such that it's sufficiently larger than a new connection handshake cost. This
+makes amplification attacks completely unattractive.
 
-In order to prevent attacks like the one described in [RFC 3489, Section
-12.1.1](https://www.rfc-editor.org/rfc/rfc3489#section-12.1.1) (see excerpt
-below), implementations MUST NOT dial any multiaddress unless it is based on the
-IP address the requesting node is observed as. This restriction as well implies
-that implementations MUST NOT accept dial requests via relayed connections as
-one can not validate the IP address of the requesting node.
+On receiving a `DialRequest`, the server selects the first address it is capable
+of dialing. If this selected address has a ip different from the clients
+observed ip. The server sends a `DialDataRequest` message with `numBytes` set to
+a sufficiently large value on the `/libp2p/autonat/2.0.0/dial-request` stream
 
-> RFC 3489 12.1.1 Attack I: DDOS Against a Target
->
-> In this case, the attacker provides a large number of clients with the same
-> faked MAPPED-ADDRESS that points to the intended target. This will trick all
-> the STUN clients into thinking that their addresses are equal to that of the
-> target. The clients then hand out that address in order to receive traffic on
-> it (for example, in SIP or H.323 messages). However, all of that traffic
-> becomes focused at the intended target. The attack can provide substantial
-> amplification, especially when used with clients that are using STUN to enable
-> multimedia applications.
+Upon receiving a `DialDataRequest` the client starts transferring `numBytes`
+number of bytes to the server.
```

Comment:
Should I add a step where the client first sends a message indicating that it has accepted/rejected the cost? 
@marten-seemann

Reply:
i added this step.